modules/subdyn/src/SD_FEM.f90

```diff
@@ -793,11 +832,44 @@ SUBROUTINE SetElementProperties(Init, p, ErrStat, ErrMsg)
       P1    = p%Elems(I, iMProp  )
       P2    = p%Elems(I, iMProp+1)
       eType = p%Elems(I, iMType)
+      COSM_ind = p%Elems(I, iMCOSMID)
 
       ! --- Properties common to all element types: L, DirCos (and Area and rho)
       Point1 = Init%Nodes(N1,2:4)
       Point2 = Init%Nodes(N2,2:4)
-      CALL GetDirCos(Point1, Point2, DirCos, L, ErrStat2, ErrMsg2); if(Failed()) return ! L and DirCos
+      Dx_=Point2(1)-Point1(1)
+      Dy_=Point2(2)-Point1(2)
+      Dz_=Point2(3)-Point1(3)
+
+      if (COSM_ind/=-1) then
+         L = real(sqrt( Dx_**2 + Dy_**2 + Dz_**2), ReKi)
+         DirCos(1, 1) =  Init%COSMs(COSM_ind, 2)
+         DirCos(1, 2) =  Init%COSMs(COSM_ind, 3)
+         DirCos(1, 3) =  Init%COSMs(COSM_ind, 4)
+         DirCos(2, 1) =  Init%COSMs(COSM_ind, 5)
+         DirCos(2, 2) =  Init%COSMs(COSM_ind, 6)
+         DirCos(2, 3) =  Init%COSMs(COSM_ind, 7)
+         DirCos(3, 1) =  Init%COSMs(COSM_ind, 8)
+         DirCos(3, 2) =  Init%COSMs(COSM_ind, 9)
+         DirCos(3, 3) =  Init%COSMs(COSM_ind, 10)
+         DirCos = TRANSPOSE(DirCos)
```

Comment:
Yes, I think we might need this transpose. I think SubDyn use an opposite convention as the rest of OpenFAST. The DirCos of OpenFAST are always from inertial to body. But SubDyn uses the transpose. Please double check. We should make sure that the documentation of OpenFAST is clear on what to put in the DirCos table

Reply:
Don’t worry the exact convention will be qualified in the tests rather than being assumed.